sdk/core/System.ClientModel/src/Internal/CancellationHelper.cs

```diff
@@ -48,4 +48,24 @@ internal static void ThrowIfCancellationRequested(CancellationToken cancellation
             ThrowOperationCanceledException(innerException: null, cancellationToken);
         }
     }
-}
\ No newline at end of file
+
+    /// <summary>Throws a cancellation exception if cancellation has been requested via <paramref name="messageToken"/> or <paramref name="timeoutToken"/>.</summary>
+    /// <param name="messageToken">The user-provided token.</param>
+    /// <param name="timeoutToken">The linked token that is cancelled on timeout provided token.</param>
+    /// <param name="innerException">The inner exception to use.</param>
+    /// <param name="timeout">The timeout used for the operation.</param>
+#pragma warning disable CA1068 // Cancellation token has to be the last parameter
+    internal static void ThrowIfCancellationRequestedOrTimeout(CancellationToken messageToken, CancellationToken timeoutToken, Exception? innerException, TimeSpan timeout)
+#pragma warning restore CA1068
+    {
+        ThrowIfCancellationRequested(messageToken);
```

Comment:
Is it intentional to ignore the `innerException` in this case?

Reply:
This is existing Azure.Core code -- if it's not easy to plumb this through quickly, I will file an issue to track

Reply:
Addressed